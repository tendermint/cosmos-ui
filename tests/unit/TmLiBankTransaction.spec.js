import { shallowMount } from "@vue/test-utils"
import { TmLiBankTransaction } from "@/index"
import transactions from "../../assets/txs.js"

describe("TmLiBankTransaction", () => {
  let wrapper
  let propsData = {
    devMode: true,
    transaction: transactions[0],
    address: "tb1d4u5zerywfjhxuc9nudvw"
  }

  beforeEach(() => {
    wrapper = shallowMount(TmLiBankTransaction, { propsData })
  })

  it("has the expected html structure", () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it("should show incoming transactions", () => {
    expect(wrapper.vm.sent).toBe(false)
  })

  it("should show outgoing transactions", () => {
    wrapper.setProps({
      transaction: transactions[1],
      address: "tb1d4u5zerywfjhxuc9nudvw"
    })
    expect(wrapper.vm.sent).toBe(true)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it("should show transactions sent to the sender", () => {
    wrapper.setProps({
      transaction: transactions[2],
      address: "tb1d4u5zerywfjhxuc9nudvw"
    })
    expect(wrapper.vm.sentSelf).toBe(true)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
