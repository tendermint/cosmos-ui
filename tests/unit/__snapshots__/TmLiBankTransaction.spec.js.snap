// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmLiBankTransaction has the expected html structure 1`] = `
<tmlitransaction-stub
  block="3436"
  color="#215C2F"
  devmode="true"
  time="2042"
/>
`;

exports[`TmLiBankTransaction should show outgoing transactions 1`] = `
<tmlitransaction-stub
  block="3438"
  color="#25293F"
  devmode="true"
  time="2052"
/>
`;

exports[`TmLiBankTransaction should show transactions sent to the sender 1`] = `
<tmlitransaction-stub
  block="466"
  color="#215C2F"
  devmode="true"
  time="1142"
/>
`;
